src/native/credentials_provider.c

```diff
@@ -52,6 +54,9 @@ static void s_on_shutdown_complete(void *user_data) {
     }
 
     (*env)->DeleteWeakGlobalRef(env, callback_data->java_crt_credentials_provider);
+    if (callback_data->jni_delegate_credential_handler != NULL) {
```

Comment:
The cleanup of the callback data needs to be refactored into a separate function and then invoked both here and in the failure case of the delegate constructor since global and weak refs could have been built at that point.  In fact it looks like none of the constructors delete the existing weak global ref (which might be safe) but it definitely isn't now that there's potentially a strong global ref.

Reply:
Thanks! Good catch! Updated🙆‍♂️